criu/proc_parse.c

```diff
@@ -853,7 +853,7 @@ int parse_pid_stat(pid_t pid, struct proc_pid_stat *s)
 	if (fd < 0)
 		return -1;
 
-	n = read(fd, buf, BUF_SIZE);
+	n = read(fd, buf, BUF_SIZE - 1);
```

Comment:
buf[BUF_SIZE] is always 0. Pls look at its declaration:

struct buffer {
        char buf[BUF_SIZE];
        char end; /* '\0' */
};

static struct buffer __buf;
static char *buf = __buf.buf;